src/elements/menu/menu-button/menu-button.visual.spec.ts

```diff
@@ -0,0 +1,96 @@
+import { html, nothing } from 'lit';
+import { repeat } from 'lit/directives/repeat.js';
+
+import {
+  describeEach,
+  describeViewports,
+  visualDiffDefault,
+  visualDiffFocus,
+  visualDiffHover,
+} from '../../core/testing/private.js';
+
+import './menu-button.js';
+
+describe(`sbb-menu-button`, () => {
+  describeViewports({ viewports: ['zero', 'medium'] }, () => {
+    const state = {
+      amount: [undefined, 123],
+      slotted: [false, true],
+    };
+
+    for (const visualDiffState of [visualDiffDefault, visualDiffHover, visualDiffFocus]) {
+      it(
+        visualDiffState.name,
+        visualDiffState.with(async (setup) => {
+          await setup.withFixture(
+            html`
+              ${repeat(
+                new Array(3),
+                (_, index) => html`
+                  <sbb-menu-button amount="123" icon-name="tick-small"
+                    >Details ${index}</sbb-menu-button
+                  >
+                `,
+              )}
+            `,
+            {
+              backgroundColor: 'var(--sbb-color-black)',
+              width: '256px',
+              focusOutlineDark: true,
+            },
+          );
+        }),
+      );
+    }
+
+    describeEach(state, ({ amount, slotted }) => {
+      it(
+        visualDiffDefault.name,
+        visualDiffDefault.with(async (setup) => {
+          await setup.withFixture(
+            html`
+              ${repeat(
+                new Array(3),
+                (_, index) => html`
+                  <sbb-menu-button amount=${amount || nothing}>
+                    Details ${index}
+                    ${slotted ? html`<sbb-icon slot="icon" name="pie-small"></sbb-icon>` : nothing}
+                  </sbb-menu-button>
+                `,
+              )}
+            `,
+            {
+              backgroundColor: 'var(--sbb-color-black)',
+              width: '256px',
+            },
+          );
+        }),
+      );
+    });
+
+    for (const disabled of [true, false]) {
+      it(
+        `long label disabled=${disabled} ${visualDiffDefault.name}`,
+        visualDiffDefault.with(async (setup) => {
```

Comment:
Maybe I'd also test hover and focus here

Reply:
disabled case has no focus/icon, but i could split in two (disabled default + long label default/hover/focus)
what do you think?

Reply:
I'd like to test that focus and hover do not have any visual effect (since it's something that we handle manually in our implementation). A sort of non-regression test